### PR TITLE
Add support for fscache

### PR DESCRIPTION
**Pull request by francescsanjuanmrf**
This PR tries to address https://github.com/prometheus/procfs/issues/259

This pull requests adds code to get and parse fscache data from /proc/fs/fscache/stats.

The parsing is based in the order of fields. I reviewed the Linux kernel code, and the output of /proc/fs/fscache/stat hasn't changed in the last 5 years..

### CODE REVIEW

**Comment by francescsanjuanmrf** [comment]
Thanks for the info!

I've made the requested changes and now the tests are OK.

File: fscache.go [hunk 1]
```diff
@@ -0,0 +1,698 @@
...
+	for s.Scan() {
+		// Each line has at least a name and value; we ignore the unit.
+		fields := strings.Fields(s.Text())
+		if len(fields) < 2 {
+			return nil, fmt.Errorf("malformed Fscacheinfo line: %q", s.Text())
+		}
+
+		switch fields[0] {
+		case "Cookies:":
+			m.index_cookies_allocated, err = strconv.ParseUint(strings.Split(fields[1], "=")[1], 0, 64)
```
Comment (pgier) [inline fscache.go]: There appears to be some risk of running into array index out of bounds exception here if there are fewer fields than expected.  I'm wondering if these cases could be refactored using a function that checks for the expected number of fields and then sets them.

```golang
func setFSCacheFields(fields []string, setFields  ...*uint64) error {
    if len(fields) < len(setFields) {
        return fmt.Errorf("Insufficient number of fields, expected %v, got %v", len(setFields), len(fields))
    }
    for i := range setFields {
        *setFields[i], err = strconv.ParseUint(strings.Split(fields[i], "=")[1], 0, 64)
        if err != nil {
            return error
        }
    }
}
```

Then for each case, you just pass the fields array and the list of fields to be set.
```golang
    err := setFSCacheFields(fields, &m.index_cookies_allocated, &m.data_storage_cookies_allocated, &m.special_cookies_allocated)
```

WDYT?

File: fscache.go [hunk 2]
```diff
@@ -0,0 +1,698 @@
...
+	// Number of stale objects deleted
+	cacheev_stale_objects_deleted uint64
+	// Number of objects retired when relinquished
+	cacheev_retired_when_relinquished uint64
+	// Number of objects culled
+	cacheev_objects_culled uint64
+}
+
+// Meminfo returns an information about current kernel/system memory statistics.
+// See https://www.kernel.org/doc/Documentation/filesystems/proc.txt
```
[the remark this answers is not in the document]
Comment (francescsanjuanmrf) [reply fscache.go]: I've changed the link to point to fscache docs.

File: fscache.go [hunk 3]
```diff
@@ -0,0 +1,698 @@
...
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/procfs/internal/util"
+)
+
+// Fscache represents memory statistics.
+type Fscacheinfo struct {
+	// Number of index cookies allocated
+	index_cookies_allocated uint64
```
[the remark this answers is not in the document]
Comment (francescsanjuanmrf) [reply fscache.go]: The fields are now camel case and start with capital letter.

File: fscache.go (hunk 1 above)
Comment (francescsanjuanmrf) [reply fscache.go]: Sure, I've changed to your suggested method, which is much safer.

**Comment by francescsanjuanmrf** [comment]
Thanks for your comments! I've added the changes.

**Comment by pgier** [comment]
@discordianfish @SuperQ any concerns about merging this?

**Comment by Opolonik** [comment]
@francescsanjuanmrf Now, I am sorry for stupid questions, but since procfs now actually has data from /proc/fs/fscache/stats, how do we collect it with node_exporter? Can't find any collectors related to fscache

**Pull request merged**